src/fdcache.cpp

```diff
@@ -276,6 +276,159 @@ bool CacheFileStat::Release()
   return true;
 }
 
+//------------------------------------------------
+// fdpage_list_t utility
+//------------------------------------------------
+// Compress the page list
+//
+// ignore_load:     Ignore the flag of loaded member and compress
+// ignore_modify:   Ignore the flag of modified member and compress
+// default_load:    loaded flag value in the list after compression when ignore_load=true
+// default_modify:  modified flag value in the list after compression when default_modify=true
+//
+// NOTE: ignore_modify and ignore_load cannot both be true.
+//
+static fdpage_list_t raw_compress_fdpage_list(const fdpage_list_t& pages, bool ignore_load, bool ignore_modify, bool default_load, bool default_modify)
+{
+  fdpage_list_t compressed_pages;
+  fdpage        tmppage;
+  bool          is_first = true;
+  for(fdpage_list_t::const_iterator iter = pages.begin(); iter != pages.end(); ++iter){
+    if(!is_first){
+      if( (!ignore_load   && (tmppage.loaded   != iter->loaded  )) ||
+          (!ignore_modify && (tmppage.modified != iter->modified)) )
+      {
+        // Different from the previous area
+        if(0 < tmppage.bytes){
+          if(ignore_load){
+            tmppage.loaded   = default_load;
```

Comment:
Extra spaces (repeated 5x).

Given that this logic is repeated, perhaps extract it to a helper function?

Reply:
I agree, and prepare the helper.

Reply:
I'll upload new code soon.